### PR TITLE
Parts 1-4

### PR DESCRIPTION
**Pull request by SleepyUser**
Pawns movement, Rook Movement, Bishops movement.

(I realised the code wasn't formatted and I've formatted them all in my current branch, had an issue with my email being on the commits, so I had to squash them once I'd updated my git email to get rid of it)

### CODE REVIEW

File: src/engine/pieces/bishop.ts [hunk 1]
```diff
@@ -1,13 +1,32 @@
...
+import Square from "../square";
 
 export default class Bishop extends Piece {
     public constructor(player: Player) {
         super(player);
     }
 
     public getAvailableMoves(board: Board) {
-        return new Array(0);
+        let availableSquares: Square[] =[];
```
Comment (kwunkiuma) [inline src/engine/pieces/bishop.ts]: This `availableSquares` can be a const! Calling push on arrays doesn't tend to count as modifying them 🙂

You should be able to update the other usages to const too!

File: src/engine/pieces/bishop.ts [hunk 2]
```diff
@@ -1,13 +1,32 @@
...
+        const square = board.findPiece(this);
+        for (let i = 1; Math.max(square.col +i, square.row + i) <= 7; i++)
```
Comment (kwunkiuma) [inline src/engine/pieces/bishop.ts]: We should be able to use && for these two conditionals too, I think that's slightly better than using max/min 🙂

**Review by kwunkiuma — CHANGES_REQUESTED** [review]
Looks good! Just some small suggestions 😄 

Opening braces on newlines is more of a C/C# convention. In TypeScript we tend to do it at the end of the same line 🙂

Real projects will have auto-formatters set up so it generally doesn't need any attention paid to it

File: src/engine/pieces/bishop.ts (hunk 1 above)
Comment (SleepyUser) [reply src/engine/pieces/bishop.ts]: I believe I got them all :)

**Comment by SleepyUser** [comment]
> Looks good! Just some small suggestions 😄
> 
> Opening braces on newlines is more of a C/C# convention. In TypeScript we tend to do it at the end of the same line 🙂
> 
> Real projects will have auto-formatters set up so it generally doesn't need any attention paid to it

I realised this right after I got into doing the next branch, as I had the "auto format when commit" turned on in FizzBuzz. In all the branches after this one I've turned that on again, so it will be fixed in the next PR.

**Review by kwunkiuma — APPROVED** [review]
LGTM, thanks!

**Pull request merged**